src/datasets/load.py

```diff
@@ -1574,46 +1584,64 @@ def dataset_module_factory(
     # Try remotely
     elif is_relative_path(path) and path.count("/") <= 1:
         try:
-            _raise_if_offline_mode_is_enabled()
-            hf_api = HfApi(config.HF_ENDPOINT)
+            # Get the Dataset Card + get the revision + check authentication all at in one call
+            # We fix the commit_hash in case there are new commits in the meantime
+            api = HfApi(
+                endpoint=config.HF_ENDPOINT,
+                token=download_config.token,
+                library_name="datasets",
+                library_version=__version__,
+                user_agent=get_datasets_user_agent(download_config.user_agent),
+            )
             try:
-                dataset_info = hf_api.dataset_info(
+                _raise_if_offline_mode_is_enabled()
```

Comment:
What happens if this line is not introduced? `huggingface_hub` is supposed to raise if `HF_HUB_OFFLINE` is set